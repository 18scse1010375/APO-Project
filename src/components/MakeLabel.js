import React from "react";
import { Input, Form, FormGroup, Label, Container, Button } from 'reactstrap';
import { ToastContainer, toast } from 'react-toastify'
import 'react-toastify/dist/ReactToastify.css';
import MakeHeader from "./MakeHeader";

const MakeLabel = () => {

  const f1 = () => toast.success("Successfully Added")

  return (
    <div>
      <MakeHeader />
      <Container className="mt-4" style={{ width: '550px' }}>

        <Form>
          <FormGroup>
            <Label for="exampleName">
              Name
            </Label>
            <Input
              id="exampleName"
              placeholder="Enter Name"
              type="text"
            />

            <Label for="exampleEmail">
              Email
            </Label>
            <Input
              id="exampleEmail"
              placeholder="Enter Email"
              type="email"
            />

            <Container className="text-center mt-4">
              <Button onClick={f1} color="primary">Submit</Button>
            </Container>

          </FormGroup>

        </Form>

        <ToastContainer />

      </Container>
    </div>
  )
}

export default MakeLabel; 
