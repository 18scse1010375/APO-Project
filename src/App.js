// import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import MakeHeader from './components/MakeHeader';
import { BrowserRouter as Router, Routes, Route,useLocation} from 'react-router-dom'
import MakeCard from './components/MakeCard';
import MakeLabel from './components/MakeLabel';
import Login from './components/Login';
import { useEffect, useState } from 'react';
import BulkEmail from './components/BulkEmail';
import 'react-toastify/dist/ReactToastify.css';
import Geeks from 'components/Geeks';
import { NoMatch } from './components/NoMatch';

import { setEnv } from 'configs';
import { adobeIms } from './services/AdobeIms';
import { userStore } from './store/UserStore';


export const LocationDisplay = () => {
  const location = useLocation()

  return <div data-testid="location-display">{location.pathname}</div>
}











function App() {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const [isLogin, setIsLogin] = useState(false);
  useEffect(() => {
    if (!userStore.isImsInit) {
      adobeIms.initialize();
      userStore.setImsInit(true);
    }

    if (adobeIms.isSignedInUser()) {
      setIsLogin(true)
    }

  }, [userStore.isImsInit]);

  useEffect(() => {
    fetch("/env")
      .then(async res => setEnv(await res.json()));
  })
  return (
    <>
    
      {/* <Router> */}
        <Routes>
          <Route exact path='/home' element={< MakeCard />}></Route>
          <Route exact path='/email-suppresion' element={< MakeLabel />}>  </Route>
          <Route exact path='/bulk-email-sent' element={< BulkEmail />}>  </Route>
          <Route exact path='/' element={<Login />} >  </Route>
          <Route exact path='/logout' element={<Login />} >  </Route> 
          <Route path="*" element={<NoMatch />} />

        </Routes>
      {/* </Router> */}

            <LocationDisplay/>

    </>
  );
}

export default App;
