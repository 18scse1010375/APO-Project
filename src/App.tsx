import { Header } from 'Header';
import './App.css';
import { adobeIms, login, updateUserProfile} from './services/AdobeIms';
import { userStore } from './store/UserStore';
import {useEffect} from 'react';

const signIn = ()=> {
  adobeIms.signIn();
}

const logout = () => {
  try {
    adobeIms.signOut();
    let profile = null;
    userStore.updateData(profile, 'token');
  } catch (e) {
    console.error('logout', e);
  }
}

function App() {

  useEffect(() => {
    if (!userStore.isImsInit) {
      adobeIms.initialize();
      userStore.setImsInit(true);
    }
  }, [userStore.isImsInit]);

  return (
    <div className="App">
      <Header store = {userStore}></Header>
      <header className="App-header">
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <button onClick={signIn}>Sign in</button>
        <button onClick={logout}>Sign out</button>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default App;
