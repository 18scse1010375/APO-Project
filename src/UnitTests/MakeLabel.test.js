/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable no-undef */
import {render, screen,fireEvent} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import axios from 'axios';

import { renderHook,act } from '@testing-library/react';

import React from 'react'
import '@testing-library/jest-dom'
import { mount, shallow } from "enzyme";

import {BrowserRouter, MemoryRouter} from 'react-router-dom'
import MakeLabel from 'components/MakeLabel'
import App from '../App';
import { wrap } from 'module';

jest.mock('components/MakeHeader');
jest.spyOn(console, "error").mockImplementation(() => {});



describe("Check the functionality of Email Suppression" ,()=>{
    const badRoute = '/email-suppresion'
    

     

     

    test("Verify <Email Suppression/> is successfully rendered or not" ,()=>{

        let wrapper=shallow(<MakeLabel/>);
        expect(wrapper).toBeDefined();
    });

    test("renders component", () => {
      const { container } = render(<MakeLabel/>);
      expect(container).toMatchSnapshot();
    });

  
  

     test("Testing for Api Integration", () => {
      const sendDataToServer = jest.fn();
      // const user = {
      //   firstName: "Mo",
      //   lastName: "Dwina",
      //   email: "b@b.com",
      //   id: "123",
      // };
      const email="singhalarun03@gmail.com"
      // const { queryByText } = render(
      //   <UserCard user={user} handleDelete={handleDelete} />
      // );


      const {queryByText} = render(
        <MemoryRouter initialEntries={[badRoute]}>
          <App />
            
        </MemoryRouter>
           )


    
      const button = queryByText("Submit");
      console.log(button)
      fireEvent.click(button);
    
     
      //  expect(sendDataToServer).toHaveBeenCalledTimes(1);
    });



    test('should throw error when axios post error', () => {
      jest.mock('axios')
      const mockedMouseEvent = {
        preventDefault: jest.fn()
      };
      const mockedError = new Error('network error');
      const wrapper = shallow(<MakeLabel/>);
      expect(wrapper.find('button')).toHaveLength(1);
      // (axios.post as jest.MockedFunction).mockRejectedValueOnce(mockedError);

      const consoleLogSpyOn = jest.spyOn(console, 'log');
      
      wrapper.find('button').simulate('click', mockedMouseEvent);
      expect(wrapper.find('button').text()).toBe('Submit');
      //  expect(axios.post).toBeCalledWith('/logout');

    })
    //   (axios.post as jest.MockedFunction<typeof axios.post>).mockRejectedValueOnce(mockedError);
    //   const consoleLogSpyOn = jest.spyOn(console, 'log');
    //   wrapper.find('button').simulate('click', mockedMouseEvent);
    //   expect(wrapper.find('button').text()).toBe('Logout');
    //   expect(axios.post).toBeCalledWith('/logout');

    //   setImmediate(() => {
    //     expect(consoleLogSpyOn).toBeCalledWith(mockedError);
    //     consoleLogSpyOn.mockRestore();
    //     done();
    //   });
    // });



















































  

    test("<MakeLabel/>" ,()=>{

        render(
            <MemoryRouter initialEntries={[badRoute]}>
              <App />
                
            </MemoryRouter>
          )
      
        
          fireEvent.mouseOver(screen.getByTestId("tooltip"));   //verification for the Tooltip title  in UI
          expect(
            screen.findByText("Please Enter Email to remove from Suppression")
          )
        
        expect(screen.getByTestId("email-input")).toBeInTheDocument()   //verification for the input box
          
        expect(screen.getByTestId("email-input")).toHaveAttribute('type', 'email')
        expect(screen.getByTestId("tooltip")).toBeInTheDocument()     //verification for the ToolTip box
        expect(screen.getByTestId("submit")).toBeInTheDocument() //Verification for the Submit Button
        expect(screen.getByTestId("submit").textContent).toEqual("Submit")   //Check the text content of Button
        expect(screen.getByTestId("submit").getAttribute("disabled")).toBe(null)  //Check Submit Button is active or not
        
   } )


    test("Backend function is calling" , ()=>{
      const sendDataToServerSpy = jest.fn();
      const badRoute="/email-suppresion"

  

      const {debug}=render(
        <MemoryRouter initialEntries={[badRoute]}>
          <App />
        </MemoryRouter>
      )

      debug();
    const inputEl = screen.getByTestId("email-input");
     userEvent.type(inputEl, "test@mail.com");
    console.log("Value are:" , document.getElementById("email-box").value)
 
    expect(screen.getByTestId("email-input")).toHaveValue("test@mail.com");  //check that i am able to pass the value in input box

    const btn=screen.getByTestId("submit")


    fireEvent.click(btn); //verification for the click event

    expect(screen.getByTestId("status"))

  //  expect(screen.getByTestId("200")).toBeInTheDocument();
    // expect(screen.getByTestId('status')).toHaveTextContent('200');


      })




  })