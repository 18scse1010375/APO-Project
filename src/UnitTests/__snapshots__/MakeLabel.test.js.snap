// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the functionality of Email Suppression renders component 1`] = `
<div>
  <div
    style="margin: auto auto auto 300px;"
  >
    <div
      class="mt-4 mx-5 container"
      style="width: 600px; display: inline-block;"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        data-testid="tooltip"
        tabindex="0"
        title="Please Enter Email to remove from Suppression"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <div
            style="width: 28px; display: inline-block; color: green; margin-top: 1px;"
          >
            <svg
              aria-hidden="true"
              class="spectrum-Icon_368b34 spectrum-Icon--sizeM_368b34"
              focusable="false"
              role="img"
              viewBox="0 0 36 36"
            >
              <path
                d="M18,2A16,16,0,1,0,34,18,16,16,0,0,0,18,2Zm-.3,4.3a2.718,2.718,0,0,1,2.864,2.824A2.664,2.664,0,0,1,17.7,11.987a2.705,2.705,0,0,1-2.864-2.864A2.717,2.717,0,0,1,17.7,6.3ZM22,27a1,1,0,0,1-1,1H15a1,1,0,0,1-1-1V25a1,1,0,0,1,1-1h1V18H15a1,1,0,0,1-1-1V15a1,1,0,0,1,1-1h4a1,1,0,0,1,1,1v9h1a1,1,0,0,1,1,1Z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="my-4 mx-1"
        style="display: inline-block; width: 450px; margin-top: 300px;"
      >
        <input
          class="form-control"
          data-testid="email-input"
          id="email-box"
          placeholder="Enter Email"
          type="email"
          value=""
        />
      </div>
      <div
        class="text-center mt-1 container"
      >
        <button
          class="btn btn-primary"
          data-testid="submit"
          id="submit"
          type="submit"
        >
          Submit
        </button>
        <div
          class="Toastify"
        />
      </div>
    </div>
    <br />
    <br />
    <span
      class="my-4 "
      id="detail"
      style="font-weight: bold; margin-left: 320px; color: red;"
    >
       
    </span>
    <h4>
       HTTP Status:
      <span
        class="my-4 "
        data-testid="status"
        id="status1"
        style="font-weight: bold; color: green;"
      />
         
    </h4>
    <br />
    <br />
    <br />
    <hr />
    <h3>
      Your Valid Input Email: 
      <span
        class="my-4 "
        id="email-id"
        style="font-weight: bold; margin-top: 200px; color: purple;"
      />
        
    </h3>
  </div>
</div>
`;
